sdk/aqueduct/decorator.py

```diff
@@ -80,6 +80,10 @@ def wrap_spec(
     if len(input_artifacts) < 1:
         raise Exception("Transformation requires at least one input artifact.")
 
+    for artifact in input_artifacts:
+        if artifact._from_flow_run:
+            raise Exception("Artifact fetched from flow run can not be reused for computation")
```

Comment:
This should be a `UserAction...Exception`. Look at the specific errors we have defined in `errors.py`